feat: Add option to group assessments by Module (#4360)
PrairieLearn/PrairieLearn: docs/course.md

    ],
```

## Units

> Each assessment in the course belongs to a *unit* defined in `infoCourse.json`. Units should be thought of as a unit of instruction, typically with a consistent number of assessments per unit, such as 3 homeworks, 2 labs, and 1 exam. Properties for assessment units are as follows.

Reviewer: "typically with a consistent number of assessments per unit, such as 3 homeworks, 2 labs, and 1 exam"

I think this sounds a bit restrictive. I'm ok with making this kind of suggestion, but in my courses I have some units that take one class (and thus less homeworks), while other units take a couple of weeks.

Property | Description
--- | ---
`name` | One word to identify the unit, lowercase.
`heading` | Longer title for the unit, displayed to users.

> 

Reviewer: I think this should move to after the example.

> Units are optional and do not affect any behavior by default. Setting the `infoCourse.json` property `"groupBy"` to `"unit"` (over the default `"set"`) will cause assessments to be grouped together by unit on the student assessments overview page. Because units are defined in an array, the order that they are defined in will be the order that they display in, top-down, to students. This can let students view their list of assessments in a chronological order, rather than simply grouped by set.

Reviewer: ```suggestion
Units are optional and do not affect any behavior by default. Setting the `infoCourse.json` property `"groupBy"` to `"unit"` (instead of the default `"set"`) will cause assessments to be grouped together by unit on the student assessments overview page. The order that they are defined in will be the order that they display in, top-down, to students. This can let students view their list of assessments in a chronological order, rather than simply grouped by set.
```

      "heading": "Quizzes",
      "color": "red1"
    },
    {
      "abbreviation": "PE",
      "name": "Practice Exam",
      "heading": "Practice Exams",
      "color": "yellow1"
    },
    {
      "abbreviation": "PQ",
      "name": "Practice Quiz",
      "heading": "Practice Quizzes",
      "color": "pink1"
    },
    {
      "abbreviation": "HW",
      "name": "Homework",
      "heading": "Homeworks",
      "color": "green1"
    },
    {
      "abbreviation": "P",
      "name": "Prep",
      "heading": "Question Preparation",
      "color": "gray1"
    },
    {
      "abbreviation": "MP",
      "name": "Machine Problem",
      "heading": "Machine Problems",
      "color": "turquoise1"
    },
    {
      "abbreviation": "WS",
      "name": "Worksheet",
      "heading": "Worksheets",
      "color": "purple1"
    }
  ]
}
```

## Assessment units

Each assessment in the course belongs to a _unit_ defined in `infoCourse.json`. Units should be thought of as a unit of instruction, or in PrairieLearn terms, a collection of assessments related to one another, but not necessarily of the same type. This means you can have a unit with two homeworks, one lab, and one exam, for example.

> Units are optional and do not affect any behavior by default. Setting the `infoCourse.json` property `"groupBy"` to `"Unit"` (instead of the default `"Set"`) will cause assessments to be grouped together by unit on the student assessments overview page. The order that they are defined in will be the order that they display in, top-down, to students. This can let students view their list of assessments in a chronological order, rather than simply grouped by set.

Reviewer 2: groupBy should not be a property of infoCourse.json, at least not only be defined there. I can see newly created courses that want to use "Units" at the start for all courses instances. However, I believe the use case for this property is the most useful at the course instance level.
